arcgis-ios-sdk-samples/Display information/Filter by time extent/FilterByTimeExtentViewController.swift

```diff
@@ -0,0 +1,126 @@
+        }
+    }
+    
+    /// The date formatter.
+    static let formatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy/MM/dd HH:mm"
+        return formatter
+    }()
+    
+    /// The data's start date.
+    let startTime = formatter.date(from: "2005/9/1 05:00")!
+    /// The data's end date.
+    let endTime = formatter.date(from: "2005/12/31 05:00")!
+    /// The default date and time for the starting thumb.
+    let currentStartTime = formatter.date(from: "2005/10/01 05:00")
+    /// The default date and time for the ending thumb.
+    let currentEndTime = formatter.date(from: "2005/10/31 05:00")
+    
+    /// The feature table URL tracking hurricanes in 2005.
+    static let featureTableURL = URL(string: "https://services5.arcgis.com/N82JbI5EYtAkuUKU/ArcGIS/rest/services/Hurricane_time_enabled_layer_2005_1_day/FeatureServer/0")!
+    
+    /// The feature layer made from the feature table.
+    let featureLayer = AGSFeatureLayer(
+        featureTable: AGSServiceFeatureTable(
+            url: featureTableURL
+        )
+    )
+    
+    /// The time slider from the ArcGIS toolkit.
+    let timeSlider = TimeSlider()
+    
+    /// Configure the time slider's attributes and position.
+    func setupTimeSlider() {
+        // Configure time slider.
+        timeSlider.labelMode = .thumbs
+        timeSlider.addTarget(self, action: #selector(timeSliderValueChanged(timeSlider:)), for: .valueChanged)
+        // Add the time slider to the view.
+        view.addSubview(timeSlider)
+        
+        // Add constraints to position the slider.
+        let margin: CGFloat = 10.0
+        timeSlider.translatesAutoresizingMaskIntoConstraints = false
+        // Set the time slider on top of the attribution bar in the map view.
+        timeSlider.bottomAnchor.constraint(equalTo: mapView.attributionTopAnchor, constant: -margin).isActive = true
+        // Set the side constraints with padding for the time slider .
+        timeSlider.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor, constant: margin).isActive = true
+        timeSlider.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor, constant: -margin).isActive = true
+    }
+    
+    /// Initialize the time slider's steps.
+    func initializeTimeStepsFromQuery() {
+        featureLayer.load { [unowned self] error in
+            guard error == nil else { fatalError(error!.localizedDescription) }
+            populateFeaturesWithQuery { _ in
+                timeSlider.initializeTimeSteps(timeStepCount: 200, fullExtent: AGSTimeExtent(startTime: startTime, endTime: endTime)) { _ in
+                    // Show the time slider.
+                    timeSlider.currentExtent = AGSTimeExtent(startTime: currentStartTime, endTime: currentEndTime)
+                    
+                    mapView.map!.operationalLayers.add(featureLayer)
+                }
```

Comment:
Nitpicking here - I think the playback speed for this sample can be faster. I'm thinking 
- timeStepCount = 60
- timeSlider.playbackInterval = 0.5

so that the sample is in a 30-second loop, that gives the user enough time to understand what is going on, while the polylines appear-disappearance still being noticeable.